src/test/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/vcf/VcfFuncotationFactoryUnitTest.java

```diff
@@ -286,4 +291,159 @@ public void testCreateFuncotationMetadata(final VariantContext variant,
+        };
+    }
+
+    /**
+     * This also tests the VCFFuncotationFactory Caching.
+     */
+    @Test(dataProvider = "provideMultiallelicTest")
+    public void testQueryIntoMultiallelic(final SimpleInterval variantInterval, final List<String> alleles,
+                                          final List<Map<String, String>> gtAttributes) {
+
+        // Make the factory
+        final VcfFuncotationFactory vcfFuncotationFactory =
+                new VcfFuncotationFactory(FACTORY_NAME, FACTORY_VERSION, IOUtils.getPath(EXAC_SNIPPET));
+
+        final ReferenceContext referenceContext = new ReferenceContext(ReferenceDataSource.of(Paths.get(FuncotatorReferenceTestUtils.retrieveB37Chr3Ref())), variantInterval);
+
+        final List<Feature> vcfFeatures;
+        try (final VCFFileReader vcfReader = new VCFFileReader(IOUtils.getPath(EXAC_SNIPPET))) {
+            vcfFeatures = vcfReader.query(variantInterval.getContig(), variantInterval.getStart(), variantInterval.getEnd()).stream().collect(Collectors.toList());
+        }
+
+        final VariantContext variant = new VariantContextBuilder()
+                .chr(variantInterval.getContig()).start(variantInterval.getStart()).stop(variantInterval.getEnd())
+                .alleles(alleles)
+                .make();
+
+        final List<Funcotation> funcotations = vcfFuncotationFactory.createFuncotationsOnVariant(
+                variant,
+                referenceContext,
+                vcfFeatures,
+                Collections.emptyList()
+        );
+        Assert.assertEquals(funcotations.size(), alleles.size() - 1);
+        Assert.assertEquals(funcotations.size(), gtAttributes.size());
+
+        IntStream.range(0, funcotations.size()).forEach( j ->
+                gtAttributes.get(j).keySet().forEach(k ->
+                        Assert.assertEquals(funcotations.get(j).getField(vcfFuncotationFactory.getName() + k), gtAttributes.get(j).get(k), "Mismatch with " + k)
+            )
+        );
+
+        Assert.assertEquals(vcfFuncotationFactory.cacheHits, 0);
+        Assert.assertEquals(vcfFuncotationFactory.cacheMisses, 1);  // Should match the number of times createFuncotationOnVariant was called.
+
+        // Funcotate again, so that we should get a cache hit.
+        final List<Funcotation> funcotations2 = vcfFuncotationFactory.createFuncotationsOnVariant(
+                variant,
+                referenceContext,
+                vcfFeatures,
+                Collections.emptyList()
+        );
+        Assert.assertEquals(vcfFuncotationFactory.cacheHits, 1);
+        Assert.assertEquals(vcfFuncotationFactory.cacheMisses, 1);
+        Assert.assertEquals(funcotations, funcotations2, "Even though there was a cache hit, the funcotations are not equal.");
+    }
+
+    @Test
```

Comment:
Can you also test the 3 arg version here as well?

Reply:
I test it at the end of the method.  Only one test should be needed, since I test that the funcotations retrievals are equal.

Done

Reply:
Ah.  I missed that call at the end.  Looks good.